core/src/client/Client.ts

```diff
@@ -47,5 +47,10 @@ interface ClientConfigurations {
   requestTimeout?: number
 }
 
+interface ExtendedClientConfiguration extends ClientConfigurations {
```

Comment:
I was hesitant about what to name this interface, so I left this as a placeholder

Reply:
ClientConfiguration & { userAgent: string }